File: apps/api-gateway/src/routes/health.ts
import { FastifyInstance } from 'fastify';

export async function healthRoutes(app: FastifyInstance) {
  // Liveness probe - is the server running?
  app.get('/health', async (request, reply) => {
    return {
      status: 'ok',
      timestamp: new Date().toISOString(),
      service: 'api-gateway',
    };
  });

  // Readiness probe - is the server ready to accept traffic?
  app.get('/ready', async (request, reply) => {
    // TODO: Add database and Redis health checks
    const checks = {
      database: true, // Placeholder - will check actual DB connection
      redis: true,    // Placeholder - will check actual Redis connection
    };

    const allHealthy = Object.values(checks).every(Boolean);

    if (!allHealthy) {
      reply.status(503);
    }

    return {
      status: allHealthy ? 'ready' : 'not_ready',
      timestamp: new Date().toISOString(),
      checks,
    };
  });
}

File: apps/ai-service/src/api/routes/health.py
from fastapi import APIRouter
from datetime import datetime

router = APIRouter()

# Health check endpoint
@router.get("/health")
async def health_check():
    """Liveness probe - is the service running?"""
    return {
        "status": "healthy",
        "service": "ai-service",
        "timestamp": datetime.utcnow().isoformat(),
    }


# Readiness check endpoint
@router.get("/ready")
async def readiness_check():
    """Readiness probe - is the service ready to accept requests?"""
    return {
        "status": "ready",
        "service": "ai-service",
        "timestamp": datetime.utcnow().isoformat(),
    }
